### PR TITLE
Add more Getting Started guides

### PR DESCRIPTION
**Pull request by rmccue**
Adds the following guides:

* Your First Theme
* Custom Modules
* Third-Party Plugins

Also updates the existing guides so there's a direct path through all docs.

### CODE REVIEW

File: other-docs/getting-started/custom-modules.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+
+### Registering Your Module
+
+Within your module's entrypoint file, your module needs to register itself with the Platform core. To do this, you need to add an action on `hm-platform.modules.init` which calls `HM\Platform\Module::register`. This function call registers the module as well as any default configuration and a callback for the module.
+
+A basic registration call looks like:
+
+```php
+use function HM\Platform\register_module;
```
Comment (roborourke) [inline other-docs/getting-started/custom-modules.md]: Probably add a namespace and opening `<?php` here so it's more copy & pastable

File: other-docs/getting-started/custom-modules.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+		],
+		__NAMESPACE__ . '\\bootstrap'
+	);
+} );
+```
+
+
+### Module Settings
+
+When registering your module, you pass various parameters to `Module::register`. These are used to control the behaviour of your module.
```
Comment (roborourke) [inline other-docs/getting-started/custom-modules.md]: `register_module()` here

File: other-docs/getting-started/custom-modules.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+
+Your module's directory is used for automated loading and parsing of documentation. The Documentation module automatically looks for a `docs` subdirectory within this directory and parses Markdown files from this.
+
+The default configuration passed in integrates with the [configuration system](configuration.md), and can be used to configure your module on different environments, or to provide an easy way to enable and disable specific functionality. Within your module, you can use `HM\Platform\get_config()` to retrieve the resolved configuration; your module configuration can be accessed as `get_config( $id )`, where `$id` is the same ID you pass to the module registration.
```
Comment (roborourke) [inline other-docs/getting-started/custom-modules.md]: `get_config()` doesn't take any params. The module config can be accessed from the module object though using `$module->get_settings()`. Should perhaps be `$module->get_config()` ...

File: other-docs/getting-started/custom-modules.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+For open-source packages, you should then publish this package to [Packagist](https://packagist.org/).
+
+For private packages you don't want to publish publically, you can use Private Packagist or Satis; we recommend following the [Composer private packages guide](https://getcomposer.org/doc/articles/handling-private-packages-with-satis.md).
+
+Once published, this can then be added to your project's `composer.json` file via the `composer require` command.
+
+
+## Next Steps
+
+Platform is built atop the open-source WordPress CMS. In addition to custom Platform modules, you can also use [third-party WordPress plugins](third-party-plugins.md) to take advantage of the open-source ecosystem.
```
Comment (roborourke) [inline other-docs/getting-started/custom-modules.md]: who says "atop"? 😛

File: other-docs/getting-started/custom-modules.md (hunk 1 above)
Comment (rmccue) [reply other-docs/getting-started/custom-modules.md]: I kept this one minimal as it's more for documenting the specifics of the function; the example below is better for copy-and-pasting.

File: other-docs/getting-started/custom-modules.md (hunk 4 above)
Comment (rmccue) [reply other-docs/getting-started/custom-modules.md]: I say it all the time! 😆

**Pull request merged**